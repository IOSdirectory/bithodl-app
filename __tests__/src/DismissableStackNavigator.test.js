import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import DismissableStackNavigator from '../../src/DismissableStackNavigator';

jest.mock('react-navigation', () => ({
  StackNavigator: () => 'StackNav'
}));

describe('DismissableStackNavigator(routes, options)', () => {
  it('is a function', () => {
    expect(typeof DismissableStackNavigator).toBe('function');
  });

  it('accepts two arguments', () => {
    expect(DismissableStackNavigator.length).toBe(2);
  });

  it('returns a component', () => {
    const DismissableStackNav = DismissableStackNavigator();
    expect(DismissableStackNav.prototype).toBeInstanceOf(Component);
  });

  describe('the component', () => {
    it('renders correctly', () => {
      const DismissableStackNav = DismissableStackNavigator();

      const navigationMock = {
        state: {
          key: '87f275fe-2f44-46d7-a5c5-e9c18913baca'
        },
        goBack: () => {}
      };

      const tree = renderer.create(
        <DismissableStackNav
          screenProps={{}}
          navigation={navigationMock}
        />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
