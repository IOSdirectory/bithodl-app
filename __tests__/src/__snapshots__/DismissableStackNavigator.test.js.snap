// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DismissableStackNavigator(routes, options) the component renders correctly 1`] = `
<StackNav
  navigation={
    Object {
      "goBack": [Function],
      "state": Object {
        "key": "87f275fe-2f44-46d7-a5c5-e9c18913baca",
      },
    }
  }
  screenProps={
    Object {
      "dismiss": [Function],
    }
  }
/>
`;
